books/spiders/books_list.py

```diff
@@ -0,0 +1,76 @@
+from dataclasses import dataclass
+
+import scrapy
+from scrapy import Selector
+from scrapy.http import Response
+from enum import Enum
+
+
+@dataclass
+class Book:
+    title: str
+    price: float
+    rating: int
+    amount_in_stock: str
+    category: str
+    description: str
+    upc: str
+
+
+class RatingEnum(Enum):
+    Zero = 0
+    One = 1
+    Two = 2
+    Three = 3
+    Four = 4
+    Five = 5
+
+
+def get_rating_value_from_class_names(class_names: str) -> int | None:
+    class_names_list = class_names.split()
+    rating_value_class = class_names_list[-1]
+    return RatingEnum[rating_value_class].value
+
+
+class BooksListSpider(scrapy.Spider):
+    name = "books_list"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/"]
+
+    def parse_book_with_details(self, response: Response) -> None:
+        book_data = response.meta["book_data"]
+        title = book_data.css("a::attr(title)").get()
+        price = book_data.css(".price_color::text").get().replace("£", "")
```

Comment:
When extracting the price, ensure that the string can always be converted to a float. Consider handling exceptions that may arise if the price format changes or is not as expected.